### ISSUE TITLE
[Flutter Web] Widgets not resizing (disappear) with browser zoom in / out

### ISSUE THREAD

**Issue by reza00farjam** [flutter/flutter#129182]
### Is there an existing issue for this?

- [x] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Open the Flutter web application in a browser.
2. Zoom in or out using the browser zoom controls.
3. Observe that some of the widgets become cutoff and disappear from the screen when zoomed too far.

P. S.
This issue was also asked on [stackoverflow](https://stackoverflow.com/questions/75409320/flutter-web-widgets-not-resizing-with-browser-zoom) 4 months ago and got no answer yet.

### Expected results

All widgets should remain fully visible and properly resized with the browser zoom, without becoming cutoff or disappearing from the screen.

### Actual results

Some of the widgets become cutoff and disappear from the screen.

### Code sample

Any flutter web code.

### Logs

No logs in terminal when this happen.

### Flutter Doctor output

[√] Flutter (Channel stable, 3.10.5, on Microsoft Windows [Version 10.0.22621.1848], locale en-US)
    • Flutter version 3.10.5 on channel stable at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (7 days ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[√] Chrome - develop for the web
    • Chrome at C:\Users\reza\AppData\Local\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.6.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.6.33801.468
    • Windows 10 SDK version 10.0.22000.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.79.2)
    • VS Code at C:\Users\reza\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.67.20230601

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.1848]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 114.0.5735.134
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 100.0.1185.36

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

**Comment by darshankawar** [comment]
Thanks for the report @reza00farjam 
I tried sample code on web (mac + chrome) and used zoom controls to zoom in and out but didn't notice the widgets resizing, as shown below:


https://github.com/flutter/flutter/assets/67046386/08c97dcd-9f8d-4d33-8e9a-6f93c2efad29

If this doesn't represent your scenario, please provide us a minimal code sample, steps to replicate and a short video that shows the reported behavior.

**Comment by agreaves** [comment]
Hi @darshankawar, we've noticed this as well. I can try to put together a reproducible example soon, but the issue is it isn't reproducible on everyone's screen because it has to do with the `devicePixelRatio`. When the browser zooms out enough, this value dips below `1.0` and this is when you'll notice the resizing issue.

On a Mac this value is typically high so you have to zoom out quite far to be able to reproduce it. It should be easier to reproduce on most Windows devices or less high quality displays.

**Comment by reza00farjam** [comment]
> Thanks for the report @reza00farjam 
> I tried sample code on web (mac + chrome) and used zoom controls to zoom in and out but didn't notice the widgets resizing, as shown below:
> 
> 
> https://github.com/flutter/flutter/assets/67046386/08c97dcd-9f8d-4d33-8e9a-6f93c2efad29
> 
> If this doesn't represent your scenario, please provide us a minimal code sample, steps to replicate and a short video that shows the reported behavior.
> 
> 

@darshankawar thanks for checking it out and sorry for late response.
I believe @agreaves sample code and additional information on how to reproduce the issue satisfying your investigation needs.
Thank you both.

**Comment by agreaves** [comment]
Just to add more detail, after some investigation the issue is caused by this commit: https://github.com/flutter/engine/commit/5bae18365a0142eb38b24d2037d3076057844953. Will try to see if I can find a workaround based on this information.

**Comment by darshankawar** [comment]
Thanks for the update and repo @agreaves. I was able to replicate this using latest stable and master:

<img width="1199" alt="Screenshot 2023-06-26 at 11 01 27 AM" src="https://github.com/flutter/flutter/assets/67046386/c4ecdf7e-3e68-48dd-b722-e569c9d68607">

This doesn't occur in previous stable release: 3.7.12:

<img width="1198" alt="Screenshot 2023-06-26 at 11 12 02 AM" src="https://github.com/flutter/flutter/assets/67046386/cb73d88d-aaa2-4fcd-848d-84272ec812f0">

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.10.5, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.10.5 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (2 days ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.12.0-10.0.pre.35, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.12.0-10.0.pre.35 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c7f82a1ca2 (77 minutes ago), 2023-06-25 23:44:09 -0400
    • Engine revision 88ff46e1ef
    • Dart version 3.1.0 (build 3.1.0-246.0.dev)
    • DevTools version 2.24.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

https://github.com/flutter/flutter/issues/96447#issuecomment-1119031780 for reference.

**Comment by VincentDR** [comment]
If more info can help.
We are having the same issue on web after upgrading to flutter 3.10.X.
Zoom of 50% or less on Chrome, MacOS with a SMS24A450 screen or with a zoom of strictly less than 50% on my native screen on my MacBookPro.
The dark blue color of the background is defined in the body of the index.html.
 
Some of our users on Windows, with the upscale resolution of 125% reported the same issue, couldn't check by myself yet.

<a href="https://ibb.co/QNbRRDQ"><img src="https://i.ibb.co/ncgvvPm/Capture-d-e-cran-2023-07-12-a-15-06-31.png" alt="Capture-d-e-cran-2023-07-12-a-15-06-31" border="0"></a>

<details>
<summary>It's better with some code:</summary>

````
@override
  Widget build(BuildContext context) {
    Size preferredSize = const Size.fromHeight(70);
    MediaQueryData mediaQueryData = MediaQuery.of(context);
    print(mediaQueryData.devicePixelRatio);
    return MaterialApp(
      home: DefaultTabController(
        length: 2,
        child: Scaffold(
          appBar: AppBar(
            toolbarHeight: preferredSize.height,
            backgroundColor: Colors.red,
            shape: Border.all(style: BorderStyle.none),
            elevation: 0,
            automaticallyImplyLeading: false,
            title: Row(
              crossAxisAlignment: CrossAxisAlignment.center,
              children: [
                Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 25),
                  child: Container(
                    color: Colors.green,
                    width: 100,
                    height: 70,
                  ),
                ),
                Expanded(
                  child: PreferredSize(
                    preferredSize: preferredSize,
                    child: SizedBox(
                      height: preferredSize.height,
                      child: TabBar(
                        tabs: [
                          Tab(
                            child: Text(
                              'Test 1',
                            ),
                          ),
                          Tab(
                            child: Text(
                              'Test 2',
                            ),
                          ),
                        ],
                      ),
                    ),
                  ),
                ),
                SizedBox(
                  height: preferredSize.height,
                  child: Row(
                    children: [
                      const VerticalDivider(
                        thickness: 1,
                        color: Colors.white,
                        indent: 20,
                        endIndent: 20,
                        width: 30,
                      ),
                      Text(
                        '{userState.user.firstName} {userState.user.lastName}',
                      ),
                    ],
                  ),
                ),
              ],
            ),
          ),
          body: TabBarView(
            children: [
              Icon(Icons.directions_transit, size: 350),
              Icon(Icons.directions_car, size: 350),
            ],
          ),
        ),
      ),
    );
  }
  ````
  </details>

<details>
<summary> Flutter doctor</summary>
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.0 22A380 darwin-x64, locale fr-FR)
    • Flutter version 3.10.5 on channel stable at /Users/--/PRO/Installs/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (4 weeks ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0-rc1)
    • Android SDK at /Users/--/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0-rc1
    • ANDROID_HOME = /Users/--/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

• No issues found!

</details>

**Comment by slavap** [comment]
@yjbanov Any progress on this? For us, it is a blocker with migration to 3.10

**Comment by gromikkool** [comment]
@yjbanov This is a critical bug for us, any updates on this?

**Comment by Sikman** [comment]
Greetings community. Will this issue be fixed in the upcoming releases? It is really critical bug, which we need to solve as soon as possible.

**Comment by slavap** [comment]
@ditman @yjbanov 
Please share with us at least some info about ETA of fixing this issue - it is a serious **blocker** for using Flutter Web. And it is present in the two latest releases: 3.10 and 3.13, also in 3.14 beta.

**Comment by yjbanov** [comment]
No update yet. I'll kick it back into triage queue.

**Comment by slavap** [comment]
@yjbanov do you mean it cannot reproduced on your side? Why put it to triage again? As I said - the bug is there in 3.10, 3.13, 3.14 - and it is a really bad one.

**Comment by yjbanov** [comment]
@slavap I put it back to triage because it looks like we did not triage it correctly.

Are you able to use the CanvasKit renderer on desktop? If you don't specify any renderer, Flutter Web picks HTML on mobile and CanvasKit on desktop.

**Comment by slavap** [comment]
@yjbanov I’m using html renderer everywhere, manually specifying it during the build. Canvaskit is getting better, but still too big for using in my projects.

**Comment by ditman** [comment]
Currently taking a look at the root cause of this.

**Comment by ditman** [comment]
Looks like some of the effects/widgets are being culled/cropped when the devicePielRatio is < 1, for example, at 0.6667:

<img width="1363" alt="Screenshot 2023-09-05 at 8 44 50 PM" src="https://github.com/flutter/flutter/assets/1255594/3b125bbc-3eb2-4128-a051-ee02edcacb96">

(Note how the size of the canvas element matches exactly where the gradient is stopping. Also note how above it there's a missing `draw-rect` for the last rectangle.)

Diving a little bit deeper.

**Comment by slavap** [comment]
@ditman 
Not sure how browser zoom may affect devicePielRatio - the problem occurs both ways, i.e. when zoom < 100% and when zoom > 100%

**Comment by ditman** [comment]
> Not sure how browser zoom may affect devicePielRatio

@slavap browser zoom (at least in Chrome) seems to be achieved by rendering the page with a different devicePixelRatio than the one of the monitor.

You can see this by calling `window.devicePixelRatio` in the JS inspector of this same page, changing the zoom level, then calling it again. (In my case, initially I get "2", but as I reduce zoom, I get a smaller value.)

What the renderer is saying above is that the "clip" rectangle for the scene is too small for what we want to render.

My current culprit is this:

* https://github.com/flutter/engine/commit/5bae18365a0142eb38b24d2037d3076057844953#diff-37400fbe08fffe96800d70edf7713084679b664a322e5af4a39f4e62a38c3b34

I'm sure we're not computing the `localClipBounds` correctly there.

**Comment by ditman** [comment]
@slavap I think I have partially solved the bug (?) please take a look and let me know what you see (I'm on a mac, and zoom in/out seem to work fine):

* https://dit-tests.web.app

**Comment by slavap** [comment]
@ditman 
I've tested on Windows 10 with Chrome and Firefox - looks ok. But once got a crash in Chrome - was not able to reproduce it though.

**Comment by ditman** [comment]
> But once got a crash in Chrome

@slavap I've seen the same crash, it seems we're attempting to use a `canvas` that is too large, and the browser refuses. I'll dive a little bit deeper. Thanks for taking a look!

**Comment by slavap** [comment]
@ditman thank you for working on fixing that bug, it is very important for everyone using Flutter Web.
Was able to catch crash few more times - it happens on 200+ zooms (300, 400, ...) only so far - so huge canvas could be an explanation.

**Comment by Junnkrd** [comment]
Here localhost works fine! In production(Edge/chrome) happens only in zoom out below 99%.

**Comment by ditman** [comment]
I still don't understand why the canvas is growing up so much. I found some code that does this, but it should never scale the canvas over 4x its intended size. The error seems to happen if the app goes below DPR 1, and then back up. It does even if I apply my fix only when DPR < 1 (and then zoom up) so there must be some problem that triggers once DPR goes below 1 (and then zooms up)!

**Comment by slavap** [comment]
@ditman Oh gosh, one problem enlights another :-( When the Flutter Web will become mature enough? (rhetoric question)

**Comment by ditman** [comment]
> When the Flutter Web will become mature enough?

Hopefully never, if Flutter Web catches up to the web, that means the web has stopped growing ;)

**Comment by slavap** [comment]
@ditman 
May be it worth to split fixing zoom and memory problems? My users are totally upset with zoom bug and pinging me daily :-(

**Comment by LucasFoulonMongai** [comment]
Hello, any update on the fix ? Seems like the only workaround is to stay on an older version of Flutter :/

**Comment by slavap** [comment]
@ditman 
Please share with us the plans for fixing this issue - it is impossible to plan Flutter Web releases without understanding when this issue will be resolved.

**Comment by Belatar** [comment]
would like to contribute as this issue is also affecting blue deck (https://bsky.app/profile/deck.blue), a tweetdeck app for bluesky that is currently gaining in popularity quickly.
if i zoom out to gain more space for columns, space on the right side of the browser screen is chopped off, making entire columns disappear. this is on a windows machine using chrome, where pixel ratio is at 1 at 100% zoom which is quite normal for windows. so the issue immediately appears when zooming out.

**Comment by slavap** [comment]
@yjbanov @ditman 
Please give us at least some highlights about fixing this issue - Flutter Web html renderer is broken for the last two stable releases 3.10 and 3.13. Are you planning to release 3.16 with a broken html renderer again? Does that mean that html renderer is deprecated silently and you are forcing us to switch to canvaskit by not fixing this issue? That is absolutely unacceptable to remain silent about such a serious bug, that way you are killing not only html renderer, but also trust in Flutter Web as a tech stack :-(

**Comment by dutsky** [comment]
Don't drop HTML-renderer, please. Its performance is far better than CanvasKit on slow (terminal) stations without decent GPU.

**Comment by gromikkool** [comment]
@yjbanov @ditman You are not giving us any information on this task, obviously quite a few people have encountered it. We would like to get an answer to the question, what are your plans for this task in order to understand what we should do next. I understand that you have a lot of tasks and work to do, but writing an answer on our problem would be great.

**Comment by ditman** [comment]
I'm sorry for the silence, we've had a few weeks of distracting, "other" type of work (meetings and stuff).

I'll try to work on this next week. We're currently working very hard to get multi-view support going on the web, and the html renderer is in the back-burner.

Please, consider using canvaskit, it's getting most of the love and optimizations lately!

**Comment by slavap** [comment]
@ditman Oh, thank god you are still there :-) Please please fix this html renderer bug - it is absolutely understandable Flutter team wishes to deprecate it, BUT canvaskit is still not suitable for mobile and low-performing devices. Another problem is that despite improvements, even chromium specific canvaskit.wasm is around 2mb gzipped, which is very bad for web apps. It has to be at least twice less before it will be ok to accept it - and I doubt that is possible to reduce size twice for that wasm.

**Comment by slavap** [comment]
@ditman Any news about fixing this issue?

**Comment by slavap** [comment]
@ditman @yjbanov 
Guys, now we have html renderer broken in the last THREE stable releases (3.10, 3.13, 3.16). Please stop silently ignoring this issue and clearly declare html renderer as broken and deprecated - it makes no sense to use it with this bug not fixed. Developers have to understand, that Flutter Web html renderer cannot be used in a production environments.

**Comment by selmaskopljakovic** [comment]
Do you happen to have any news here? We need this fixed as soon as possible.

**Comment by joshuajylin** [comment]
There is a non-english charactors rendering issue if not using html renderer, so I cannot stop using html renderer.
#138869

**Comment by phamnhuvu-dev** [comment]
Our customers are angry already 😢

**Comment by slavap** [comment]
@ditman @yjbanov 
Please calm us all down and provide any info about fixing this issue - it is already too intense, I cannot believe that obvious P1 issue could not be fixed for 5 months :-(

**Comment by QinGuiCheng** [comment]
Do you happen to have any news here? We need this fixed as soon as possible.

**Comment by QinGuiCheng** [comment]
> @slavap我想我已经部​​分解决了这个错误（？）请看一下并让我知道您看到了什么（我在 Mac 上，放大/缩小似乎工作正常）：
> 
> * https://dit-tests.web.app

**Comment by QinGuiCheng** [comment]
> @slavap我想我已经部​​分解决了这个错误（？）请看一下并让我知道您的情况（我在 Mac 上，放大/缩小似乎工作正常）：
> 
> * https://dit-tests.web.app

Is this connection packaging using Flutter build web -- web renderer HTML? Could you please provide the demo and packaging steps?

**Comment by ditman** [comment]
I'll put the "solution" I found above in a branch, and share it with @yjbanov so we can figure out a version of it that doesn't crash the browser :)

**Comment by slavap** [comment]
@yjbanov 
Please assign someone to this issue - it needs to be fixed, that is absolutely critical for many of us.

**Comment by QinGuiCheng** [comment]
Do you have a solution?

**Comment by phamnhuvu-dev** [comment]
Our customers are angry already 😢, it is reality, not fake
<img width="701" alt="image" src="https://github.com/flutter/flutter/assets/22906656/4d810ec0-e795-4412-9c78-48f86c709993">

**Comment by slavap** [comment]
@yjbanov @ditman 
Please assign someone to this issue and fix it - it is impossible to continue using Flutter Web in prod with this bug not fixed.

**Comment by cuonghoang96** [comment]
Any news on this issue? It's quite a long time and nothing happened. At least show us a progress on this issue.

**Comment by capibarbara** [comment]
@jmagman @christopherfujino  
Is this something you can help with? This is causing a lot of issues for Flutter developers. Thank you!

**Comment by ditman** [comment]
A fix for this issue is incoming (see pull request linked above).

Thanks all for your patience!

**Comment by slavap** [comment]
@ditman 
Thank you so much for fixing that issue!

**Comment by ditman** [comment]
My code has broken some Flutter Goldens, I'm trying to figure out what's wrong to reland.

**Comment by slavap** [comment]
@ditman please keep us posted. By the way, what is it - "Flutter Goldens" ?

**Comment by NicolasDionB** [comment]
> @ditman please keep us posted. By the way, what is it - "Flutter Goldens" ?

I can answer that one, goldens are used in tests to compare the UI with a reference image (called golden). So if there is even a slight change in the rendering, when running the tests the comparison will fail. You can read more about them [here](https://medium.com/appunite-edu-collection/intro-to-golden-tests-in-flutter-3d23bb12d056).

**Comment by ditman** [comment]
I've posted a 2nd PR attempting to land the fix. Check the PR description for all the details!

**Comment by priyankarawat953** [comment]
Just upgraded to v3.19 today. Issue doesn't seem to be addressed. Can anyone else confirm this?

**Comment by ditman** [comment]
Apologies for not getting back to the ticket after it closed.

The [fix](https://github.com/flutter/flutter/commit/5f1a3c16bd29d747a7c56a949d426b5c17fa3b66) is present from version: `3.20.0-5.0.pre`

**Comment by slavap** [comment]
@yjbanov @ditman 
Please cherry pick this fix to STABLE version, we cannot wait till 3.22 will be released.

**Comment by ditman** [comment]
@slavap have you verified that the code currently in `master` fixes your issue?
